redisson/src/main/java/org/redisson/misc/LogHelper.java

```diff
@@ -0,0 +1,81 @@
+/**
+ * Copyright 2016 Nikita Koksharov
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.redisson.misc;
+
+import java.lang.reflect.Array;
+import java.util.Collection;
+import java.util.Objects;
+
+/**
+ * @author Philipp Marx
+ */
+public class LogHelper {
+
+    private static final int MAX_COLLECTION_LOG_SIZE = Integer.valueOf(System.getProperty("redisson.maxCollectionLogSize", "10"));
+    private static final int MAX_STRING_LOG_SIZE = Integer.valueOf(System.getProperty("redisson.maxStringLogSize", "100"));
+
+    private LogHelper() {
+    }
+
+    public static <T> String toString(T object) {
+        if (object == null) {
+            return "null";
+        } else if (object instanceof String) {
+            return toStringString((String) object);
+        } else if (object.getClass().isArray()) {
+            return toArrayString(object);
+        } else if (object instanceof Collection) {
+            return toCollectionString((Collection<?>) object);
+        } else {
+            return Objects.toString(object);
+        }
+    }
+
+    private static String toStringString(String string) {
+        if (string.length() > MAX_STRING_LOG_SIZE) {
+            return new StringBuilder(string.substring(0, MAX_STRING_LOG_SIZE)).append("...").toString();
+        } else {
+            return string;
+        }
+    }
+
+    private static String toCollectionString(Collection<?> collection) {
+        return toArrayString(collection.toArray(new Object[collection.size()]));
```

Comment:
(TBH it was also just convenient to reuse the toString method for arrays which I had implemented first :) but I am fine with adding a more sophisticated approach for the Collections/Maps if you like 👍  )